tests/models/auto/test_image_processing_auto.py

```diff
@@ -0,0 +1,167 @@
+            self.assertIsInstance(config, CLIPImageProcessor)
+
+    def test_image_processor_from_local_directory_from_config(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            model_config = CLIPConfig()
+
+            # Create a dummy config file with image_proceesor_type
+            processor_tmpfile = Path(tmpdirname) / "preprocessor_config.json"
+            model_tmpfile = Path(tmpdirname) / "config.json"
+            json.dump(
+                {"image_processor_type": "CLIPImageProcessor", "processor_class": "CLIPProcessor"},
+                open(processor_tmpfile, "w"),
+            )
+            json.dump({"model_type": "clip"}, open(model_tmpfile, "w"))
+
+            # remove image_processor_type to make sure config.json alone is enough to load image processor locally
+            config_dict = AutoImageProcessor.from_pretrained(tmpdirname).to_dict()
+
+            config_dict.pop("image_processor_type")
+            config = CLIPImageProcessor(**config_dict)
+
+            # save in new folder
+            model_config.save_pretrained(tmpdirname)
+            config.save_pretrained(tmpdirname)
+
+            config = AutoImageProcessor.from_pretrained(tmpdirname)
+
+            # make sure private variable is not incorrectly saved
+            dict_as_saved = json.loads(config.to_json_string())
+            self.assertTrue("_processor_class" not in dict_as_saved)
+
+        self.assertIsInstance(config, CLIPImageProcessor)
+
+    def test_image_processor_from_local_file(self):
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            processor_tmpfile = Path(tmpdirname) / "preprocessor_config.json"
+            json.dump(
+                {"image_processor_type": "CLIPImageProcessor", "processor_class": "CLIPProcessor"},
+                open(processor_tmpfile, "w"),
+            )
+
+            config = AutoImageProcessor.from_pretrained(processor_tmpfile)
+            self.assertIsInstance(config, CLIPImageProcessor)
+
+    def test_repo_not_found(self):
+        with self.assertRaisesRegex(
+            EnvironmentError, "clip-base is not a local folder and is not a valid model identifier"
+        ):
+            _ = AutoImageProcessor.from_pretrained("clip-base")
+
+    def test_revision_not_found(self):
+        with self.assertRaisesRegex(
+            EnvironmentError, r"aaaaaa is not a valid git identifier \(branch name, tag name or commit id\)"
+        ):
+            _ = AutoImageProcessor.from_pretrained(DUMMY_UNKNOWN_IDENTIFIER, revision="aaaaaa")
+
+    def test_image_processor_not_found(self):
+        with self.assertRaisesRegex(
+            EnvironmentError,
+            "hf-internal-testing/config-no-model does not appear to have a file named preprocessor_config.json.",
+        ):
+            _ = AutoImageProcessor.from_pretrained("hf-internal-testing/config-no-model")
+
+    def test_from_pretrained_dynamic_image_processor(self):
```

Comment:
I don't think we can use `hf-internal-testing/test_dynamic_feature_extractor` directly, as it's for wav2vec2. I could add an equivalent vision feature extractor under e.g. `hf-internal-testing/test_dynamic_feature_extractor_vision` ?

Reply:
Ah yes, good point. Adding a new repo sounds fine!

Reply:
This is a bit tricky because of the renaming that happens in `AutoImageProcessor.from_pretrained` from `FeatureExtractor` -> `ImageProcessor` in [this line](https://github.com/amyeroberts/transformers/blob/aa431f2b88672ddcfe000571229e3988dc92e185/src/transformers/models/auto/image_processing_auto.py#L296).

It tries to import `NewImageProcessor` which doesn't exist in https://huggingface.co/hf-internal-testing/test_dynamic_feature_extractor_vision/blob/main/feature_extractor.py

So either: 
* https://huggingface.co/hf-internal-testing/test_dynamic_feature_extractor_vision/blob/main/feature_extractor.py has `class NewImageProcessor(CLIPFeatureExtractor)`
* There's some logic which doesn't rename if the image processor isn't `IMAGE_PROCESSOR_MAPPING_NAMES` - although this leads loading a feature extractor, which seems like unwanted behaviour. 

What do you think the intended behaviour should be?

Reply:
Or we can just ignore custom code for feature extractors (I don't think there are any out for now). It shouldn't slow down the PR as it is a edge case we can deal with later!